libs/core/src/lib/action-sheet/action-sheet-mobile/action-sheet-mobile.component.ts

```diff
@@ -0,0 +1,52 @@
+import {
+    Component,
+    ElementRef,
+    Inject,
+    Input,
+    Optional,
+    TemplateRef,
+    ViewChild
+} from '@angular/core';
+import { DialogService } from '../../dialog/dialog-service/dialog.service';
+import { ACTION_SHEET_COMPONENT, ActionSheetInterface } from '../action-sheet.interface';
+import {
+    MOBILE_MODE_CONFIG,
+    MobileModeBase,
+    MobileModeControl,
+    MobileModeConfigToken
+} from '../../utils/base-class/mobile-mode.class';
+
+@Component({
+    selector: 'fd-action-sheet-mobile',
+    templateUrl: './action-sheet-mobile.component.html'
+})
+export class ActionSheetMobileComponent extends MobileModeBase<ActionSheetInterface> {
+
+    /** Whenever links should be visible **/
+    @Input()
+    open = false;
+
+    /** @hidden */
+    @ViewChild('dialogTemplate') dialogTemplate: TemplateRef<any>;
+
+    /** @hidden */
+    childContent: {
+        actionSheetBodyTemplate: TemplateRef<any>,
+    } = null;
+
+    constructor(
+        elementRef: ElementRef,
+        dialogService: DialogService,
+        @Inject(ACTION_SHEET_COMPONENT) actionSheetComponent: ActionSheetInterface,
+        @Optional() @Inject(MOBILE_MODE_CONFIG) mobileModes: MobileModeConfigToken[]
+    ) {
+        super(elementRef, dialogService, actionSheetComponent, MobileModeControl.ACTION_SHEET, mobileModes);
```

Comment:
I don't think we should officially announce this component as a component with "Mobile" version as for other mobile controls. This does not use the Dialog. Lets leave this for later iteration.
Right now I'd suggest to remove it from `MobileModeControl` enum and pass here `null` value.